### ISSUE TITLE
Share.share never returns when the dialog is cancelled on iOS

### ISSUE THREAD

**Issue by waynebloss** [expo/expo#8218]
## Bug Report

You can see my code in #7991 but essentially, I am calling `Share.share`, the share dialog is showing and if I cancel it, my code never gets a returned promise.

### Environment

```sh
Expo CLI 3.20.1 environment info:
    System:
      OS: Linux 4.19 Manjaro Linux
      Shell: 5.0.16 - /bin/bash
    Binaries:
      Node: 13.13.0 - /usr/bin/node
      Yarn: 1.22.4 - /usr/bin/yarn
      npm: 6.14.4 - /usr/bin/npm
    IDEs:
      Android Studio: 3.6 AI-192.7142.36.36.6392135
    npmPackages:
      expo: ~36.0.0 => 36.0.2 
      react: ~16.9.0 => 16.9.0 
      react-dom: ~16.9.0 => 16.9.0 
      react-native: https://github.com/expo/react-native/archive/sdk-36.0.1.tar.gz => 0.61.4 
      react-native-web: ~0.11.7 => 0.11.7 
    npmGlobalPackages:
      expo-cli: 3.20.1
```

### Steps to Reproduce

`await Share.share({ url, message, title })`

### Expected Behavior

Expected a resolved or rejected promise.

### Actual Behavior

The promise never resolves or rejects.

### Reproducible Demo

`await Share.share({ url, message, title })` in code and cancel share dialog in iOS.

**Comment by waynebloss** [comment]
Also, if anyone knows why the code in expo/react-native is so old?

I was looking for the native code for the ActionSheetManager to see if something was fixed recently - and I see the code here is 2 - 4 years old and tens of thousands of commits behind Facebook React Native code...

https://github.com/expo/react-native/tree/exp-latest/Libraries/ActionSheetIOS

**Comment by waynebloss** [comment]
This code looks like it might never return if `activityError` and `completed` are both false:

https://github.com/expo/react-native/commit/4b1206c2ae39516c5028f0628d99ac62ee33b2ec#diff-b6ec6c60921763a2cc180dd8ed24ba35

Full file here

https://github.com/expo/react-native/blob/4b1206c2ae39516c5028f0628d99ac62ee33b2ec/Libraries/ActionSheetIOS/RCTActionSheetManager.m#L167

**Comment by waynebloss** [comment]
You can see someone had the same issue in facebook/react-native#26809

It was fixed in facebook/react-native#26842

And it should be available in react-native v0.62 according to a comment there.

**Comment by byCedric** [comment]
Hi @waynebloss! Thanks for the report and the research! I think SDK 37 is still on `0.61.4`, I'm not sure if they backported this to older React Native versions. We will definitely include this when we upgrade to `0.62.x`, but I'm not sure how easy this would be backporting this to SDK 37. Let's keep it open until we resolve it 😄
